File: HW1_PS2_hobrecht_d.py
# Dane Hobrecht
# 1103521
# This program calculates the net gain/loss based on given stock values, among other variables.

# Input
stock_name = (input('Enter Stock name: '))
number_of_shares = int(input('Enter Number of shares : '))
purchase_price = float(input('Enter Purchase price : '))
selling_price = float(input('Enter selling price : '))
commission = float(input('Enter Commission : '))

# Processing & calculations
stock_paid = (number_of_shares * purchase_price)
commission_paid = (stock_paid * commission)
stock_sold = (number_of_shares * selling_price)
commission_sold = (stock_sold * commission)
net = (stock_sold - commission_paid) - (stock_paid + commission_sold)

# Output
print('\nAmount paid for the stock:          $' + format(stock_paid, '13,.2f'))
print('Commission paid on the purchase:    $' + format(commission_paid, '13,.2f'))
print('Amount the stock sold for:          $' + format(stock_sold, '13,.2f'))
print('Commission paid on the sale:        $' + format(commission_sold, '13,.2f'))
print('Profit (or loss if negative):       $' + format(net, '13,.2f'))

##Test Run 1
##Enter Stock name: Kaplack, Inc.
##Enter Number of shares : 10000
##Enter Purchase price : 33.92
##Enter selling price : 35.92
##Enter Commission : 0.04
##
##
##Amount paid for the stock:          $   339,200.00
##Commission paid on the purchase:    $    13,568.00
##Amount the stock sold for:          $   359,200.00
##Commission paid on the sale:        $    14,368.00
##Profit (or loss if negative):       $    -7,936.00
##
##Test Run 2
##Enter Stock name: IBM
##Enter Number of shares : 15000
##Enter Purchase price : 50.25
##Enter selling price : 100.20
##Enter Commission : 0.02
##
##
##Amount paid for the stock:          $   753,750.00
##Commission paid on the purchase:    $    15,075.00
##Amount the stock sold for:          $ 1,503,000.00
##Commission paid on the sale:        $    30,060.00
##Profit (or loss if negative):       $   704,115.00
##
##Test Run 3
##Enter Stock name: Loke, Inc.
##Enter Number of shares : 50000
##Enter Purchase price : 100.93
##Enter selling price : 147.28
##Enter Commission : 0.06 
##
##
##Amount paid for the stock:          $ 5,046,500.00
##Commission paid on the purchase:    $   302,790.00
##Amount the stock sold for:          $ 7,364,000.00
##Commission paid on the sale:        $   441,840.00
##Profit (or loss if negative):       $ 1,572,870.00
##
##Test Run 4
##Enter Stock name: Meta, Inc.
##Enter Number of shares : 2
##Enter Purchase price : 310.41
##Enter selling price : 6.66
##Enter Commission : 0.03
##
##
##Amount paid for the stock:          $       620.82
##Commission paid on the purchase:    $        18.62
##Amount the stock sold for:          $        13.32
##Commission paid on the sale:        $         0.40
##Profit (or loss if negative):       $      -626.52
##
##Test Run 5
##Enter Stock name: Red Hat, Inc.
##Enter Number of shares : 23852
##Enter Purchase price : 187.71
##Enter selling price : 283.13
##Enter Commission : 0.09
##
##
##Amount paid for the stock:          $ 4,477,258.92
##Commission paid on the purchase:    $   402,953.30
##Amount the stock sold for:          $ 6,753,216.76
##Commission paid on the sale:        $   607,789.51
##Profit (or loss if negative):       $ 1,265,215.03

File: HW1_PS2_hobrecht_d.c
/*********************************************************/
/* Name: Dane Hobrecht                                   */
/* Student ID: 1103521                                   */
/* Date: 02/01/22                                        */
/* Homework 1 Program Set 2                              */
/* This program maintains information about stocks.      */
/*********************************************************/

#include <stdio.h>

int main()
{
	/* Constants */
	char STAR_LINE[35] = "*********************************\n";

	/* Standard declarations */
	char stock_name[35];
	int number_of_shares;
	float buy_price, current_price, fees;
	float initial_cost, current_cost, profit;
	float temp_profit, total_profit;

	/* STOCK 1 */

	/* Input */
	printf("%s            Stock 1\n%s", STAR_LINE, STAR_LINE);
	printf("Enter stock name ");
	gets(stock_name);
	printf("Enter number of shares ");
	scanf("%i", &number_of_shares);
	printf("Enter the buy price, current price, fees ");
	scanf("%f %f %f", &buy_price, &current_price, &fees);

	/* Processing & calculations */
	initial_cost = number_of_shares * buy_price;
	current_cost = number_of_shares * current_price;
	profit = current_cost - initial_cost - fees;
	total_profit = profit;

	/* Output */
	printf("\nThe Stock name is    : %s", stock_name);
	printf("\nThe number of shares :  %10i", number_of_shares);
	printf("\nThe buy price is     : $%10.2f", buy_price);
	printf("\nThe current price is : $%10.2f", current_price);
	printf("\nThe fees are         : $%10.2f\n", fees);

	printf("\nThe initial cost is :  $%10.2f", initial_cost);
	printf("\nThe current cost is :  $%10.2f", current_cost);
	printf("\nThe profit is       :  $%10.2f\n", profit);

	/* STOCK 2 */

	/* Input */
	printf("%s            Stock 2\n%s", STAR_LINE, STAR_LINE);
	getchar();
	printf("Enter stock name ");
	gets(stock_name);
	printf("Enter number of shares ");
	scanf("%i", &number_of_shares);
	printf("Enter the buy price, current price, fees ");
	scanf("%f %f %f", &buy_price, &current_price, &fees);
	
	/* Processing & calculations */
	initial_cost = number_of_shares * buy_price;
	current_cost = number_of_shares * current_price;
	profit = current_cost - initial_cost - fees;
	temp_profit = profit;
	total_profit = total_profit + temp_profit;

	/* Output */
	printf("\nThe Stock name is    : %s", stock_name);
	printf("\nThe number of shares :  %10i", number_of_shares);
	printf("\nThe buy price is     : $%10.2f", buy_price);
	printf("\nThe current price is : $%10.2f", current_price);
	printf("\nThe fees are         : $%10.2f\n", fees);

	printf("\nThe initial cost is :  $%10.2f", initial_cost);
	printf("\nThe current cost is :  $%10.2f", current_cost);
	printf("\nThe profit is       :  $%10.2f\n", profit);
	
	/* STOCK 3 */

	/* Input */
	printf("%s            Stock 3\n%s", STAR_LINE, STAR_LINE);
	getchar();
	printf("Enter stock name ");
	gets(stock_name);
	printf("Enter number of shares ");
	scanf("%i", &number_of_shares);
	printf("Enter the buy price, current price, fees ");
	scanf("%f %f %f", &buy_price, &current_price, &fees);
	
	/* Processing & calculations */
	initial_cost = number_of_shares * buy_price;
	current_cost = number_of_shares * current_price;
	profit = current_cost - initial_cost - fees;
	temp_profit = profit;
	total_profit = total_profit + temp_profit;

	/* Output */
	printf("\nThe Stock name is    : %s", stock_name);
	printf("\nThe number of shares :  %10i", number_of_shares);
	printf("\nThe buy price is     : $%10.2f", buy_price);
	printf("\nThe current price is : $%10.2f", current_price);
	printf("\nThe fees are         : $%10.2f\n", fees);

	printf("\nThe initial cost is :  $%10.2f", initial_cost);
	printf("\nThe current cost is :  $%10.2f", current_cost);
	printf("\nThe profit is       :  $%10.2f\n", profit);

	/* STOCK 4 */

	/* Input */
	printf("%s            Stock 4\n%s", STAR_LINE, STAR_LINE);
	getchar();
	printf("Enter stock name ");
	gets(stock_name);
	printf("Enter number of shares ");
	scanf("%i", &number_of_shares);
	printf("Enter the buy price, current price, fees ");
	scanf("%f %f %f", &buy_price, &current_price, &fees);
	
	/* Processing & calculations */
	initial_cost = number_of_shares * buy_price;
	current_cost = number_of_shares * current_price;
	profit = current_cost - initial_cost - fees;
	temp_profit = profit;
	total_profit = total_profit + temp_profit;

	/* Output */
	printf("\nThe Stock name is    : %s", stock_name);
	printf("\nThe number of shares :  %10i", number_of_shares);
	printf("\nThe buy price is     : $%10.2f", buy_price);
	printf("\nThe current price is : $%10.2f", current_price);
	printf("\nThe fees are         : $%10.2f\n", fees);

	printf("\nThe initial cost is :  $%10.2f", initial_cost);
	printf("\nThe current cost is :  $%10.2f", current_cost);
	printf("\nThe profit is       :  $%10.2f\n", profit);

	/* STOCK 5 */

	/* Input */
	printf("%s            Stock 5\n%s", STAR_LINE, STAR_LINE);
	getchar();
	printf("Enter stock name ");
	gets(stock_name);
	printf("Enter number of shares ");
	scanf("%i", &number_of_shares);
	printf("Enter the buy price, current price, fees ");
	scanf("%f %f %f", &buy_price, &current_price, &fees);
	
	/* Processing & calculations */
	initial_cost = number_of_shares * buy_price;
	current_cost = number_of_shares * current_price;
	profit = current_cost - initial_cost - fees;
	temp_profit = profit;
	total_profit = total_profit + temp_profit;

	/* Output */
	printf("\nThe Stock name is    : %s", stock_name);
	printf("\nThe number of shares :  %10i", number_of_shares);
	printf("\nThe buy price is     : $%10.2f", buy_price);
	printf("\nThe current price is : $%10.2f", current_price);
	printf("\nThe fees are         : $%10.2f\n", fees);

	printf("\nThe initial cost is :  $%10.2f", initial_cost);
	printf("\nThe current cost is :  $%10.2f", current_cost);
	printf("\nThe profit is       :  $%10.2f\n", profit);

	printf("\nThe total profit for the 5 stocks is : $%10.2f\n", total_profit);

	return 0;
}

/*
*********************************
            Stock 1
*********************************
Enter stock name IBM CORP 
Enter number of shares 155
Enter the buy price, current price, fees 15.33 13.33 5.00

The Stock name is    : IBM CORP
The number of shares :         155
The buy price is     : $     15.33
The current price is : $     13.33
The fees are         : $      5.00

The initial cost is :  $   2376.15
The current cost is :  $   2066.15
The profit is       :  $   -315.00
*********************************
            Stock 2
*********************************
Enter stock name ORACLE SYS
Enter number of shares 375
Enter the buy price, current price, fees 11.77 12.25 3.50

The Stock name is    : ORACLE SYS
The number of shares :         375
The buy price is     : $     11.77
The current price is : $     12.25
The fees are         : $      3.50

The initial cost is :  $   4413.75
The current cost is :  $   4593.75
The profit is       :  $    176.50
*********************************
            Stock 3
*********************************
Enter stock name SUN MICRO
Enter number of shares 350
Enter the buy price, current price, fees 27.55 35.75 12.25 

The Stock name is    : SUN MICRO
The number of shares :         350
The buy price is     : $     27.55
The current price is : $     35.75
The fees are         : $     12.25

The initial cost is :  $   9642.50
The current cost is :  $  12512.50
The profit is       :  $   2857.75
*********************************
            Stock 4
*********************************
Enter stock name LINKSYS INC
Enter number of shares 85
Enter the buy price, current price, fees 25.35 23.34 6.00

The Stock name is    : LINKSYS INC
The number of shares :          85
The buy price is     : $     25.35
The current price is : $     23.34
The fees are         : $      6.00

The initial cost is :  $   2154.75
The current cost is :  $   1983.90
The profit is       :  $   -176.85
*********************************
            Stock 5
*********************************
Enter stock name CISCO INC
Enter number of shares 50
Enter the buy price, current price, fees 45.36 50.86 1.50

The Stock name is    : CISCO INC
The number of shares :          50
The buy price is     : $     45.36
The current price is : $     50.86
The fees are         : $      1.50

The initial cost is :  $   2268.00
The current cost is :  $   2543.00
The profit is       :  $    273.50

The total profit for the 5 stocks is : $   2815.90
*/
